app/models/ability.rb

```diff
@@ -0,0 +1,36 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the passed in user here. For example:
+    #
+    user ||= User.new # guest user (not logged in)
+
+    if user.present?
+      can :manage, :all
+    elsif can :manage, Post, author_id: user.id
+      can :manage, Comment, author_id: user.id
+      can :read, :all
+    else
+      can :read, :all
+    end
```

Comment:
- [ ]  Instead of checking if the user is present, this method should check if the user is an admin.  Remember:
- A user can delete a post if it is theirs or if they have an admin role
- A user can delete a comment if it is theirs or if they have an admin role

Example:

```ruby
class Ability
 include CanCan::Ability

     def initialize(user)

         user ||= User.new # guest user (not logged in)

         if user.admin? 
         can :manage, :all
         else
         can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
         can :read, :all
         end
     end
 end
```